.github/workflows/test-code.yml

```diff
@@ -70,5 +32,35 @@ jobs:
       - name: Check Code
         run: cargo check
 
-      - name: Test Code
-        run: cargo test
+  test-code:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/shared
+
+      - name: Remove rust-toolchain.toml
+        # To make sure that the nightly version will be used all throughout
+        run: |
+          rm /home/runner/work/pendulum/pendulum/rust-toolchain.toml
+
+      - name: Setup nightly Rust toolchain
+        uses:  dtolnay/rust-toolchain@nightly
+        with:
+          toolchain: nightly-2024-04-18
+          components: rustfmt, clippy
+          target: wasm32-unknown-unknown
+
+      - name: Setup nightly Rust as default
+        run: rustup default nightly-2024-04-18
+
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
somehow, accessing the secrets in the action file will cause an error. I had to call it here, on both jobs: **build-check** and **test-code**.